### PR TITLE
Add doc string to `PageIterator`

### PR DESCRIPTION
**Pull request by dlm6693**
Addressing https://github.com/boto/botocore/issues/2396

While ideally we would add `next` functionality to `PageIterator` or simply update the name of the class to prevent confusion, both would cause breaking changes. As a result all we really can do is add a doc string with additional info. Perhaps using a warning here would be warranted, but decided against it.

Let me know if there's anything that should be added to the doc string.

### CODE REVIEW

**Review by nateprewitt — CHANGES_REQUESTED** [review]
Minor clarification and formatting tweaks. Otherwise, looks good!

File: botocore/paginate.py [hunk 1]
```diff
@@ -185,6 +185,9 @@ def get_paginator(self, operation_name):
 
 
 class PageIterator:
+    """An iterable object to pagiante API results.
+    Please note it is NOT a python iterator"""
+
```
Comment (nateprewitt) [inline botocore/paginate.py:190]:
```suggestion
    """An iterable object to pagiante API results.
    
    Please note it is NOT a python iterator. Use ``iter`` to wrap this as a generator.
    """
```

Comment (dlm6693) [reply botocore/paginate.py]: done

File: botocore/paginate.py [hunk 2]
```diff
@@ -186,7 +186,8 @@ def get_paginator(self, operation_name):
 
 class PageIterator:
     """An iterable object to pagiante API results.
-    Please note it is NOT a python iterator"""
+    Please note it is NOT a python iterator.
+    Use ``iter`` to wrap this as a generator"""
```
Comment (nateprewitt) [inline botocore/paginate.py:190]: Can we make sure to include a period at the end of the last sentence? We'll also want to make sure the closing `"""` is on its own line.

Comment (nateprewitt) [reply botocore/paginate.py:190]: No we can keep it on the same line. The example in the first comment should have what's considered standard formatting for doc strings.

While we don't adhere strictly to PEP8 anymore, the [conventions for docstrings](https://peps.python.org/pep-0008/#documentation-strings) are still widely used. Generally to the effect of:

```python
class MyClass:
    """This is my class.
    
    Extra info beyond the class description.
    """
```

Comment (dlm6693) [reply botocore/paginate.py:190]: Got it thanks! Pushing up now

**Review by nateprewitt — APPROVED** [review]

**Pull request merged**